File: backend/api/apps/clubs/models/enums.py
from django.db import models

class Visibility(models.TextChoices):
    PUBLIC = 'public', 'Public'
    PRIVATE = 'private', 'Private'
    SECRET = 'secret', 'Secret'

class ClubStatus(models.TextChoices):
    ACTIVE = "active", "Active"
    PAUSED = "paused", "Paused"
    ARCHIVED = "archived", "Archived"
    SUSPENDED = "suspended", "Suspended"

class JoinMode(models.TextChoices):
    INSTANT = "instant", "Instant"
    APPLICATION = "application", "Application Required"
    INVITE_ONLY = "invite_only", "Invite Only"

class AffiliateStatus(models.TextChoices):
    PENDING = 'pending', 'Pending'
    ACTIVE = 'active', 'Active'
    REJECTED = 'rejected', 'Rejected'
    LEFT = 'left', 'Left'
    BANNED = 'banned', 'Banned'

class MembershipScope(models.TextChoices):
    GLOBAL = 'global', 'Global'
    EXCLUSIVE = 'exclusive', 'Exclusive'
    CROSS_INSTITUTE = 'cross_institute', 'Cross Institute'

File: packages/validation/src/club/enums.ts
export const StatusOptions = ["active", "paused", "archived", "suspended"] as const;
export const PrivacyOptions = ["public", "private", "secret"] as const;
export const ScopeOptions = ["global", "exclusive", "cross_institute"] as const;
export const JoinModeOptions = ["instant", "application", "invite_only"] as const;


export enum Status {
    Active = "active",
    Paused = "paused",
    Archived = "archived",
    Suspended = "suspended",
}

export enum Privacy {
    Public = "public",
    Private = "private",
    Secret = "secret",
}

export enum Scope {
    Global = "global",
    Exclusive = "exclusive",
    CrossInstitute = "cross_institute",
}

export enum JoinMode {
    Instant = "instant",
    Application = "application",
    InviteOnly = "invite_only",
}
